packages/botonic-nlp/src/tasks/intent-classification/botonic-intent-classifier.ts

```diff
@@ -20,40 +21,44 @@ import {
 import { Intent, PredictionProcessor, Processor } from './process'
 import { IntentClassificationConfigStorage } from './storage'
 
+export interface IntentClassifierConfig extends NlpTaskConfig {
+  intents: string[]
+}
+
 export class BotonicIntentClassifier {
+  readonly locale: Locale
+  readonly maxLength: number
+  readonly vocabulary: string[]
   readonly intents: string[]
   private processor: Processor
   private predictionProcessor: PredictionProcessor
   private modelManager: ModelManager
 
-  constructor(
-    readonly locale: Locale,
-    readonly maxLength: number,
-    intents: string[],
-    readonly vocabulary: string[],
-    private readonly preprocessor: Preprocessor
-  ) {
-    this.intents = unique(intents)
+  constructor(config: IntentClassifierConfig) {
+    this.locale = config.locale
+    this.maxLength = config.maxLength
+    this.vocabulary = config.vocabulary
+    this.intents = unique(config.intents)
     this.processor = new Processor(
-      this.preprocessor,
+      config.preprocessor,
       new LabelEncoder(new IndexedItems(this.vocabulary)),
       new OneHotEncoder(new IndexedItems(this.intents))
     )
-    this.predictionProcessor = new PredictionProcessor(this.intents)
+    this.predictionProcessor = new PredictionProcessor(config.intents)
   }
 
   static async load(
     path: string,
     preprocessor: Preprocessor
   ): Promise<BotonicIntentClassifier> {
     const config = new IntentClassificationConfigStorage().load(path)
```

Comment:
I am aware that I have a similar implementation of configuration in `BotonicIntentClassification` and `BotonicNer` for when the NLP task is initialized by loading a previously trained model. **This would be solved in another PR.** ⚠️